debezium-core/src/main/java/io/debezium/relational/TableSchemaBuilder.java

```diff
@@ -191,6 +193,22 @@ else if (Strings.isNullOrEmpty(tableId.schema())) {
         return null;
     }
 
+    private void validateIncomingRowToInternalMetadata(int[] recordIndexes, Field[] fields, ValueConverter[] converters,
+            Object[] row, int position) {
+        if (position >= converters.length) {
+            LOGGER.error("Error requesting a converter, converters: {}, requested index: {}", converters.length, position);
+            throw new ConnectException("Column indexing array is larger than number of converters, internal schema represantion is probably out of sync with real database schema");
+        }
+        if (position >= fields.length) {
+            LOGGER.error("Error requesting a field, fields: {}, requested index: {}", fields.length, position);
+            throw new ConnectException("Too few schema fields, internal schema represantion is probably out of sync with real database schema");
+        }
+        if (recordIndexes[position] >= row.length) {
+            LOGGER.error("Error requesting a row value, row: {}, requested index: {} at position {}", row.length, recordIndexes[position], position);
+            throw new ConnectException("Data row is smaller than a column index, internal schema represantion is probably out of sync with real database schema");
```

Comment:
s/represantion/represention/

Reply:
@gunnarmorling The check should just throw an error in case the index if out of array bounds. If it is not the case then conditions should be rfixed.

Reply:
@gunnarmorling typo fixed